src/sagemaker/jumpstart/utils.py

```diff
@@ -178,7 +189,7 @@ def get_tag_value(tag_key: str, tag_array: List[Dict[str, str]]) -> str:
     Raises:
         KeyError: If the number of matches for the ``tag_key`` is not equal to 1.
     """
-    tag_values = [tag[tag_key] for tag in tag_array if tag_key in tag]
+    tag_values = [tag["Value"] for tag in tag_array if tag_key == tag["Key"]]
     if len(tag_values) != 1:
         raise KeyError(
```

Comment:
question: should this be a `ValueError` in this case?

Reply:
i think `KeyError` makes more sense, since you're basically trying to index the array with a bad key

Reply:
+1